rpcs3/Emu/RSX/rsx_utils.h

```diff
@@ -752,6 +752,26 @@ namespace rsx
 		return result;
 	}
 
+	static inline f32 decode_fx13(u32 bits)
+	{
+		// Classic fixed point, see PGRAPH section of nouveau docs for TEX_FILTER (lod_bias) and TEX_CONTROL (min_lod, max_lod)
+		// Technically min/max lod are fixed 4.8 but a 5.8 decoder should work just as well since sign bit is 0
+
+		f32 base;
+		if ((bits & (1 << 12)) == 0)
+		{
+			base = f32(bits >> 8);
+		}
+		else
+		{
+			bits = (~bits + 1) & 0xfff;
+			base = -f32(bits >> 8);
+		}
+
+		const auto frc = (bits & 0xff) / 256.f;
+		return base + frc;
```

Comment:
if base is negative, frc should be substructed from base?

Reply:
ie is -0.5 just 0.5? frc doesnt respect thd sign.

Reply:
actually Im wrong here, because the integer part is in 2's complement you can't represent -0.
So to represent -0.5 you need -1 + 0.5.